go/lint-project.sh

```diff
@@ -117,7 +117,7 @@ if [[ "$OS_NAME" != "windows" ]]; then
     ./bin/nancy --clean-cache
 
     # Ignore Consul and Vault Enterprise, they need a gocloud.dev release
-    go list -mod=mod -m all | ./bin/nancy --skip-update-check --loud sleuth --exclude-vulnerability "$ignored"
+    go list -json -deps | ./bin/nancy --skip-update-check --loud sleuth --exclude-vulnerability "$ignored"
```

Comment:
I don't think this works. It appears that nancy is no longer scanning dependencies. 

Before: https://github.com/moov-io/infra/runs/5757619808?check_suite_focus=true#step:7:88
This Branch: https://github.com/moov-io/infra/runs/5838120901?check_suite_focus=true#step:7:64

Before:
![Screen Shot 2022-04-05 at 12 26 50 PM](https://user-images.githubusercontent.com/120951/161814986-56a9171b-14d9-4986-8d3d-fccaa3a65b83.png)

This PR: 
![Screen Shot 2022-04-05 at 12 26 48 PM](https://user-images.githubusercontent.com/120951/161814973-db00c016-e59c-4f22-91a5-11227f83951f.png)

Reply:
We probably want to investigate a different set of flags for `go list`.

https://github.com/sonatype-nexus-community/nancy/issues/228